### PR TITLE
fix: do not override all rustflags in circleci

### PR DESCRIPTION
**Pull request by crepererum**
As we've learned in #8048 and #8052, rustflags do NOT stack. Hence `--cfg tokio_unstable` gets lost in our test exeuction. Since we only want to change one specific parameter (the debug feature), use the env variable that cargo provides us.

**In contrast to the linked PRs, this only changes the test excution. Prod builds remain untouched.**

Extracted from #8050.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -86,7 +86,7 @@ jobs:
       CARGO_INCREMENTAL: "0"
       # Disable full debug symbol generation to speed up CI build
       # "1" means line tables only, which is useful for panic tracebacks.
-      RUSTFLAGS: "-C debuginfo=1"
+      DEBUG: "1"
```
Comment (domodwyer) [inline .circleci/config.yml:89]: What uses the `DEBUG` env var? 

I see https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts shows the `DEBUG` var, but it seems to be set _by cargo_ during builds, so not sure how this fits together.

Comment (crepererum) [reply .circleci/config.yml:89]: From that page:

> `OPT_LEVEL`, `DEBUG` — values of the corresponding variables for the profile currently being built.

and

> `CARGO_PROFILE_<name>_DEBUG` — What kind of debug info to include, see [profile.<name>.debug](https://doc.rust-lang.org/cargo/reference/config.html#profilenamedebug).

on that new page:

> **`profile.<name>.debug`**
>
> - Type: integer or boolean
> - Default: See profile docs.
> - Environment: `CARGO_PROFILE_<name>_DEBUG`
>
> See [debug](https://doc.rust-lang.org/cargo/reference/profiles.html#debug).

On that that page:

> **debug**
>
> The `debug` setting controls the [`-C debuginfo` flag](https://doc.rust-lang.org/rustc/codegen-options/index.html#debuginfo) which controls the amount of debug information included in the compiled binary.
>
> The valid options are:
> - `0` or `false`: no debug info at all
> - `1`: line tables only
> - `2` or `true`: full debug info

So it ultimately does the same as the `RUSTFLAGS` env prior to this PR, just that we don't override the entire RUSTFLAGS list in https://github.com/influxdata/influxdb_iox/blob/main/.cargo/config (esp. under Linux where we set a few helpful options).

Comment (domodwyer) [reply .circleci/config.yml:89]: So I made a new bin that just panics, and ran it with `RUSTFLAGS="-C debuginfo=1"`:

```shellsession
% RUSTFLAGS="-C debuginfo=1" cr; du  /Users/dom/Documents/rust/panic/target/debug/panic
   Compiling panic v0.1.0 (/Users/dom/Documents/rust/panic)
    Finished dev [unoptimized + debuginfo] target(s) in 0.15s
     Running `target/debug/panic`
thread 'main' panicked at 'Hello, world!', src/main.rs:2:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/std/src/panicking.rs:578:5
   1: core::panicking::panic_fmt
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/panicking.rs:67:14
   2: panic::main
             at ./src/main.rs:2:5
   3: core::ops::function::FnOnce::call_once
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
472	/Users/dom/Documents/rust/panic/target/debug/panic
```

So 472 bytes for the bin with reduced debug symbols. When I do `DEBUG=1` it doesn't seem to produce the same result:

```shellsession
% DEBUG=1 cr; du  /Users/dom/Documents/rust/panic/target/debug/panic
   Compiling panic v0.1.0 (/Users/dom/Documents/rust/panic)
    Finished dev [unoptimized + debuginfo] target(s) in 0.10s
     Running `target/debug/panic`
thread 'main' panicked at 'Hello, world!', src/main.rs:2:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/std/src/panicking.rs:578:5
   1: core::panicking::panic_fmt
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/panicking.rs:67:14
   2: panic::main
             at ./src/main.rs:2:5
   3: core::ops::function::FnOnce::call_once
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
1392	/Users/dom/Documents/rust/panic/target/debug/panic
```

Note that it's 1392 bytes, which is the same as a "normal" build:

```shellsession
% cr; du  /Users/dom/Documents/rust/panic/target/debug/panic
    Finished dev [unoptimized + debuginfo] target(s) in 0.00s
     Running `target/debug/panic`
thread 'main' panicked at 'Hello, world!', src/main.rs:2:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/std/src/panicking.rs:578:5
   1: core::panicking::panic_fmt
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/panicking.rs:67:14
   2: panic::main
             at ./src/main.rs:2:5
   3: core::ops::function::FnOnce::call_once
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
1392	/Users/dom/Documents/rust/panic/target/debug/panic
```

Comment (crepererum) [reply .circleci/config.yml:89]: Will investigate next week.

Comment (crepererum) [reply .circleci/config.yml]: I appearently misread the docs: `DEBUG` is only passed to build scripts, it's not read by cargo itself. However I've found another variable that works:

## Test

Create empty test project:

```console
$ cargo new cargo-debug-flag
```

Test environment variables:

```console
$ cargo clean && env <ENV> cargo build && du ./target/debug/cargo-debug-flag
```

e.g.

```console
$ cargo clean && env RUSTFLAGS='-C debuginfo=0' cargo build && du ./target/debug/cargo-debug-flag
```

| Env                         | Size |
| --------------------------- | ---- |
|                             | 4196 |
| `RUSTFLAGS='-C debuginfo=0'`| 4184 |
| `RUSTFLAGS='-C debuginfo=1'`| 4192 |
| `RUSTFLAGS='-C debuginfo=2'`| 4196 |
| `CARGO_PROFILE_DEV_DEBUG=0` | 4184 |
| `CARGO_PROFILE_DEV_DEBUG=1` | 4192 |
| `CARGO_PROFILE_DEV_DEBUG=2` | 4196 |
| `DEBUG=0`                   | 4196 |
| `DEBUG=1`                   | 4196 |
| `DEBUG=2`                   | 4196 |

Outcome:

- `RUSTFLAGS='-C debuginfo=X` works as expected
- `CARGO_PROFILE_DEV_DEBUG=X` works as expected
- `DEBUG` doesn't seem to do anything

**Review by NGA-TRAN — APPROVED** [review]
The changes in circle CI dev build

**Pull request merged**